org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/ex/ExternalAntlrLexerFragment.java

```diff
@@ -0,0 +1,177 @@
+/*******************************************************************************
+ * Copyright (c) 2019 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.xtext.generator.parser.antlr.ex;
+
+import org.eclipse.xtend2.lib.StringConcatenationClient;
+import org.eclipse.xtext.parser.antlr.Lexer;
+import org.eclipse.xtext.xtext.generator.Issues;
+import org.eclipse.xtext.xtext.generator.model.GuiceModuleAccess;
+import org.eclipse.xtext.xtext.generator.model.IXtextGeneratorFileSystemAccess;
+import org.eclipse.xtext.xtext.generator.model.TypeReference;
+import org.eclipse.xtext.xtext.generator.parser.antlr.AbstractAntlrGeneratorFragment2;
+
+import com.google.common.annotations.Beta;
+
+/**
```

Comment:
Some description what this is actually good for would be nice. I actually don't have a clue.

Reply:
i am not sure if we want users encourage to use it

Reply:
@cdietrich  We could write something like:
`Use this fragment if you want to use an external lexer , e.g: in case the generated lexer has to be customized.`

Reply:
sure. but this should be done only if people know what they do and understand what this is internal api

Reply:
Yes, this is an advanced topic.

Reply:
But without any comment it is even more advanced. Take Tamas' comment.

Reply:
:(

Reply:
I think `in case the generated lexer has to be customized.` leaves more open questions than answered questions. Why "has the lexer" to be customized? Why would I want to do that? What is an external lexer at all?